### PR TITLE
fix: add learn more link for export as skill 

### LINKED ISSUE #7851: Export as skill is missing a learn more link
## Describe the bug

The learn more link is missing from the allowed callers page of the "Export as skill" modal. See below:

Screen Shot 2021-05-17 at 3 55 55 PM

## Version

2.0.0-test

### PR DESCRIPTION
**Pull request by alanlong9278**
## Description

fix: add learn more link for select trigger, select dialog & allow callers pages in `export as skill` experience.
link
 

## Task Item

closes #7851

### CODE REVIEW

File: Composer/packages/client/src/pages/design/exportSkillModal/constants.tsx [hunk 1]
```diff
@@ -275,8 +289,19 @@ export const editorSteps: { [key in ManifestEditorSteps]: EditorStep } = {
     content: SelectDialogs,
     editJson: false,
     subText: () =>
-      formatMessage(
-        'The capabilities of your bot are defined in its dialogs and triggers. Selected dialogs will be included in the manifest. Internal dialogs or actions may not be relevant to other bots. Learn more.'
```
Comment (tdurnford) [inline Composer/packages/client/src/pages/design/exportSkillModal/constants.tsx:279]: Unless this has changed, you should be able to set the `helpLink` property, and it should take care of the link for you. If it has changed, please update the type to reflect that. 

https://github.com/microsoft/BotFramework-Composer/blob/1990c8076249aeb49209a1b7c99794c258edf17e/Composer/packages/client/src/pages/design/exportSkillModal/constants.tsx#L122

File: Composer/packages/client/src/pages/design/exportSkillModal/constants.tsx [hunk 2]
```diff
@@ -243,8 +246,19 @@ export const editorSteps: { [key in ManifestEditorSteps]: EditorStep } = {
...
     subText: () =>
-      formatMessage(
-        'To ensure a secure connection, provide the App ID of the bots that can connect to your skill.  If you don’t have this information, you can also add this information in Skill Configuration. Learn more.'
+      formatMessage.rich(
+        'To ensure a secure connection, provide the App ID of the bots that can connect to your skill.  If you don’t have this information, you can also add this information in Skill Configuration. <a>Learn more.</a>',
+        {
+          a: ({ children }) => (
+            <Link
+              key="export-skill-learn-more"
+              href={'https://docs.microsoft.com/en-us/composer/how-to-publish-skill?tabs=v2x'}
```
[the remark this answers is not in the document]
Comment (hatpick) [reply Composer/packages/client/src/pages/design/exportSkillModal/constants.tsx:255]: this url is being used multiple times, just put it in a variable and reuse it, so when it comes to updating it we dont have to update multiple locations and potentially miss some.

File: Composer/packages/client/src/pages/design/exportSkillModal/constants.tsx [hunk 3]
```diff
@@ -1,11 +1,14 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
```
Comment (hatpick) [inline Composer/packages/client/src/pages/design/exportSkillModal/constants.tsx:5]: just import React, this comes at a cost that is not necessary.

**Review by hatpick — CHANGES_REQUESTED** [review]

File: Composer/packages/client/src/pages/design/exportSkillModal/constants.tsx (hunk 3 above)
Comment (alanlong9278) [reply Composer/packages/client/src/pages/design/exportSkillModal/constants.tsx]: Refactor

File: Composer/packages/client/src/pages/design/exportSkillModal/constants.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (alanlong9278) [reply Composer/packages/client/src/pages/design/exportSkillModal/constants.tsx]: Use `helpLink` property to replace it.

File: Composer/packages/client/src/pages/design/exportSkillModal/constants.tsx (hunk 1 above)
Comment (alanlong9278) [reply Composer/packages/client/src/pages/design/exportSkillModal/constants.tsx:279]: Use `helpLink`

**Comment by hatpick** [comment]
why is there a blank space below the buttons in this dialog?

File: Composer/packages/client/src/pages/design/exportSkillModal/constants.tsx [hunk 4]
```diff
@@ -240,9 +240,10 @@ export const editorSteps: { [key in ManifestEditorSteps]: EditorStep } = {
     content: AddCallers,
     subText: () =>
       formatMessage(
-        'To ensure a secure connection, provide the App ID of the bots that can connect to your skill.  If you don’t have this information, you can also add this information in Skill Configuration. Learn more.'
+        'To ensure a secure connection, provide the App ID of the bots that can connect to your skill.  If you don’t have this information, you can also add this information in Skill Configuration.'
       ),
     title: () => formatMessage('Which bots can connect to this skill?'),
+    helpLink: 'https://docs.microsoft.com/en-us/composer/how-to-publish-skill?tabs=v2x',
```
Comment (tdurnford) [inline Composer/packages/client/src/pages/design/exportSkillModal/constants.tsx:246]: Make this an aka.ms link. Also, if it’s used multiple times, move it into a variable.

**Review by tdurnford — CHANGES_REQUESTED** [review]

File: Composer/packages/client/src/pages/design/exportSkillModal/constants.tsx (hunk 4 above)
Comment (alanlong9278) [reply Composer/packages/client/src/pages/design/exportSkillModal/constants.tsx]: Fixed it.

**Review by tdurnford — DISMISSED** [review]

**Comment by hatpick** [comment]
> > why is there a blank space below the buttons in this dialog?
> 
> Sorry, I dont get u.

In the screen recording you have attached, there is a gap between the dialog buttons and the bottom of the dialog, why is that?
Screen Shot 2021-06-07 at 1 45 21 PM
Screen Shot 2021-06-07 at 1 46 27 PM

**Comment by hatpick** [comment]
@alanlong9278 what's the status of work on this PR, did you see my question above with screenshots?

**Comment by alanlong9278** [comment]
> @alanlong9278 what's the status of work on this PR, did you see my question above with screenshots?

Sorry, I am on vacations these days. I will fix it today @hatpick

**Comment by luhan2017** [comment]
@hatpick could you please take another look at this PR?

**Comment by hatpick** [comment]
@alanlong9278 can you update your screenshots?

**Review by hatpick — APPROVED** [review]

**Pull request merged**